### PR TITLE
Refactoring + Merging features from sprint 1

### PR DESCRIPTION
**Pull request by vrooomer**
This big PR comes from the branch that we used for the demo, that merged all the work in a single activity: `WelcomeActivity`.
We decided to delete `MainActivity` and `GreetingActivity` as they won't be useful anymore.

Here is how the buttons in the new "main" activity (`WelcomeActivity`) work so far, we can discuss it later:
- Join Game: join a game with group id
- Create Game: nothing

At the bottom, from left to right:
- Button 1: rules
- Button 2: map
- Button 3: profile
- Button 4: settings

Also, to clean up our files, I created different packages: All the classes that concern the menu of the game are now in `com.github.polypoly.app.menu` and all the non-activity classes related to the menu are in `com.github.polypoly.app.menu.kotlin`

===========

Finally, while checking the tests, I saw that `ProfileActivity` is barely tested (check `WelcomeActivityTest` for inspiration). Same for the activities related to the map and the DB back-end (let me know what parts are untestable)

As this PR is quite big, I would like to get more than one review before we change to this version!

~ v:)

### CODE REVIEW

File: app/src/androidTest/java/com/github/polypoly/app/WelcomeActivityTest.kt [hunk 1]
```diff
@@ -17,7 +21,7 @@ class WelcomeActivityTest {
     private val optionButtonRules = composeTestRule.onNodeWithContentDescription("Show Rules")
     private val optionButton2 = composeTestRule.onNodeWithContentDescription("optionButton2")
```
Comment (Jharaxus) [inline app/src/androidTest/java/com/github/polypoly/app/WelcomeActivityTest.kt:22]: UI component should have a more meaningful name and id to express their intent

Comment (Jharaxus) [reply app/src/androidTest/java/com/github/polypoly/app/WelcomeActivityTest.kt:22]: This works also for optionButton3

Comment (Jharaxus) [reply app/src/androidTest/java/com/github/polypoly/app/WelcomeActivityTest.kt:22]: Why are they call **Option**Button btw?

File: app/src/main/java/com/github/polypoly/app/menu/JoinGroupActivity.kt [hunk 2]
```diff
@@ -155,8 +156,8 @@ class JoinGroupActivity : ComponentActivity() {
      * This function launches the group room activity and passes the group code to it.
      */
     private fun joinGroupRoom(mContext : Context) {
-        val groupIntent = Intent(mContext, GreetingActivity::class.java)
-        groupIntent.putExtra("groupCode", groupCode)
+        //val groupIntent = Intent(mContext, GreetingActivity::class.java)
+        //groupIntent.putExtra("groupCode", groupCode)
```
Comment (Jharaxus) [inline app/src/main/java/com/github/polypoly/app/menu/JoinGroupActivity.kt:161]: The commented code should be removed from PR

File: app/src/main/java/com/github/polypoly/app/menu/kotlin/GameMusic.kt [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    fun setVolume(value: Float) {
+        var validValue = max(value, 1f)
+        validValue = min(validValue, 0f)
+        mediaPlayer.setVolume(validValue, validValue)
+    }
+
+    fun mute() {
+        setVolume(0f)
+    }
+
```
Comment (AmLibra) [inline app/src/main/java/com/github/polypoly/app/menu/kotlin/GameMusic.kt:26]: Please avoid useless whitespaces.

File: app/src/main/java/com/github/polypoly/app/menu/kotlin/GameMusic.kt [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    private lateinit var mediaPlayer: MediaPlayer
+
+    fun startSong() {
+        mediaPlayer = MediaPlayer.create(context, songId)
+        mediaPlayer.isLooping = true
+        mediaPlayer.start()
+    }
+
+    fun setVolume(value: Float) {
+        var validValue = max(value, 1f)
```
Comment (AmLibra) [inline app/src/main/java/com/github/polypoly/app/menu/kotlin/GameMusic.kt:18]: Avoid Magic numbers!

File: app/src/main/java/com/github/polypoly/app/menu/WelcomeActivity.kt [hunk 5]
```diff
@@ -160,7 +173,9 @@ class WelcomeActivity : ComponentActivity() {
             ) {
                 Surface(
                     color = MaterialTheme.colors.primary,
-                    modifier = Modifier.fillMaxWidth(0.95f).fillMaxHeight(0.95f)
+                    modifier = Modifier
+                        .fillMaxWidth(0.95f)
+                        .fillMaxHeight(0.95f)
```
Comment (Jharaxus) [inline app/src/main/java/com/github/polypoly/app/menu/WelcomeActivity.kt:178]: Magic numbers should be avoided

File: app/src/main/java/com/github/polypoly/app/menu/WelcomeActivity.kt [hunk 6]
```diff
@@ -160,7 +173,9 @@ class WelcomeActivity : ComponentActivity() {
             ) {
                 Surface(
                     color = MaterialTheme.colors.primary,
-                    modifier = Modifier.fillMaxWidth(0.95f).fillMaxHeight(0.95f)
+                    modifier = Modifier
```
Comment (AmLibra) [inline app/src/main/java/com/github/polypoly/app/menu/WelcomeActivity.kt:176]: Avoid Magic numbers!

File: app/src/main/java/com/github/polypoly/app/menu/kotlin/GameMusic.kt [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+        validValue = min(validValue, 0f)
```
Comment (Jharaxus) [inline app/src/main/java/com/github/polypoly/app/menu/kotlin/GameMusic.kt:19]: You can use `clamp` to bind validValue between 0f ans 1f

**Review by Jharaxus — CHANGES_REQUESTED** [review]
Some minor things should be changed, but the overall merge is good :)

File: app/src/main/java/com/github/polypoly/app/menu/ProfileModifyingActivity.kt [hunk 8]
```diff
@@ -126,6 +126,7 @@ class ProfileModifyingActivity : ComponentActivity() {
 
     /**
      * The field where the player can write his/her description
```
Comment (AmLibra) [inline app/src/main/java/com/github/polypoly/app/menu/ProfileModifyingActivity.kt:128]: Documentation like this is very good, less pointless // prepended comments would be better !

**Review by AmLibra — CHANGES_REQUESTED** [review]
This seems to be a good agglomeration of everyone's work, but could use a bit of refactoring and rigor.

File: app/src/androidTest/java/com/github/polypoly/app/WelcomeActivityTest.kt (hunk 1 above)
Comment (vrooomer) [reply app/src/androidTest/java/com/github/polypoly/app/WelcomeActivityTest.kt:22]: I agree with you, the problem is that they don't have a fixed utility so far, I put some launching intents but only for the demo.
We need to decide what button does what, I'm gonna change that

File: app/src/main/java/com/github/polypoly/app/menu/kotlin/GameMusic.kt (hunk 7 above)
Comment (vrooomer) [reply app/src/main/java/com/github/polypoly/app/menu/kotlin/GameMusic.kt:19]: Nice! I didn't know about this, thanks :)

File: app/src/main/java/com/github/polypoly/app/menu/WelcomeActivity.kt (hunk 5 above)
Comment (vrooomer) [reply app/src/main/java/com/github/polypoly/app/menu/WelcomeActivity.kt:178]: I also agree with that for magic numbers. But in this case these are just parameters used locally, I don't think using a constant there should be a good idea as it depends on each composable.

**Comment by vrooomer** [comment]
I fixed most of the issues. I just disagree with the "magic number" issue, as it is only a parameter given to a unique composable and not global at all.

~ v:)

**Review by AmLibra — APPROVED** [review]

File: app/src/androidTest/java/com/github/polypoly/app/WelcomeActivityTest.kt (hunk 1 above)
Comment (Jharaxus) [reply app/src/androidTest/java/com/github/polypoly/app/WelcomeActivityTest.kt:22]: Great, this looks fine now that you added details regarding what option they refer to

File: app/src/main/java/com/github/polypoly/app/menu/WelcomeActivity.kt (hunk 5 above)
Comment (Jharaxus) [reply app/src/main/java/com/github/polypoly/app/menu/WelcomeActivity.kt:164]: Ok we discussed this together and I aggree with your point saying that it is a more global issue than just this PR, therefore I created a specific [issue](https://github.com/polypoly-team/android-app/issues/40) that we can address in a later sprint.

File: app/src/main/java/com/github/polypoly/app/menu/kotlin/GameMusic.kt (hunk 7 above)
Comment (Jharaxus) [reply app/src/main/java/com/github/polypoly/app/menu/kotlin/GameMusic.kt]: No problem :)

**Review by Jharaxus — APPROVED** [review]

**Pull request merged**